### PR TITLE
disable kickstart in ctfe to workaround Issue 16626

### PR DESCRIPTION
**Pull request by MartinNowak**
- consumes too much memory, introduced by
  e98fa4ad5ad39487844c91357cfec4f698e88230 (#4286)

### CODE REVIEW

**Comment by wilzbach** [comment]
Don't worry about the CodeCov messages. Due to [Issue 16397](https://issues.dlang.org/show_bug.cgi?id=16397) the numbers for running the entire test suite are wrong and we need to run the coverage analysis per module.

**Comment by MartinNowak** [comment]
Going to merge this so we can reenable the Higgs tests. It was fairly similar to @DmitryOlshansky's initial fix attempt and disabling the kickstarter should only make the Regex slower, but not break it.

**Comment by DmitryOlshansky** [comment]
Thanks! 
Sadly this means ctRegex is slower then normal regex in most cases. Anyhow until compiler is better at CTFE we are out of options.

**Comment by nordlow** [comment]
We just need to wait for @uplinkcoder to finish newCTFE. 😊

**Comment by nordlow** [comment]
https://github.com/UplinkCoder/dmd/tree/newCTFE?files=1

**Comment by JackStouffer** [comment]
This PR caused a regression: https://issues.dlang.org/show_bug.cgi?id=17161

**Comment by UplinkCoder** [comment]
@JackStouffer I can indeed reproduce this,  leading me to believe that the kick-starter is meant to be build at compile-time.
We cannot enable this as long as it crashes the compiler.

**Comment by JackStouffer** [comment]
But we can temporarily revert to the way things were done in 2.072.2. Which we should do ASAP, as CTFE regex is one of the poster-children for D. Hell, Andrei makes a point to mention how fast it's supposed to be every talk he does on D.

**Comment by DmitryOlshansky** [comment]
> But we can temporarily revert to the way things were done in 2.072.2. Which we should do ASAP, as CTFE regex is one of the poster-children for D. 

For the record - it works, just consumes more memory then typical tester boxes seem to have.
About the revert - the PR to get new engine was a massive and long overdue rework of std.regex internals. I was waiting for years to enable const Regex! Postponing it for another half a year (or more) is not something I'm looking forward to.

I'm being held back by the quality of CTFE ever since 2011. I'm afraid that without a NAGGING ISSUE that we MUST improve the state of CTFE nothing will be done in our time. Humiliation could be a decent motivator. 

>Hell, Andrei makes a point to mention how fast it's supposed to be every talk he does on D.

In fact in following a discussion of std.regex state with Andrei I prepared a next patch that brings ctRegex to the new level of performance. I even tested it against top competitors (with JITs and whatnot) and verified that we indeed can be *the* fastest on Earth. Guess what? Even more memory hungry due to CTFE.

TL;DR: We will get nothing while CTFE is a piece of junk. I'm accepting this regression and do not think that freezing std.regex development until CTFE is improved is a good idea.

**Comment by UplinkCoder** [comment]
@DmitryOlshansky 
I am improving CTFE as fast as I can, but it will take a few months until it can support std.regex.
If your test-boxes cannot compile the projects they are supposed to test this is a problem.

**Comment by JackStouffer** [comment]
@DmitryOlshansky Then develop it in a separate branch. In the mean time we can revert the whole regex folder to the v2.072.2 branch and performance and our image are both saved.

**Comment by DmitryOlshansky** [comment]
@UplinkCoder Thanks for stepping up and doing all the hard work! A few months sounds way more inspiring the yet another year or so.

@JackStouffer You are probably right, I just have to muster my patience meanwhile.
